### PR TITLE
form clean up

### PR DESCRIPTION
**Pull request by SabMueller**
After the last review I decided to clean up my form a bit by outsourcing parts of it (OrganizationForm.js and AnimalForm.js), adding some more validation functions and created a success view, to which the user is redirected after successfully submitting the form. Thank you for reviewing ✨

### CODE REVIEW

File: client/src/lib/Validation.js [hunk 1]
```diff
@@ -4,18 +4,36 @@ const validateEmail = (email) =>
     email
   );
 
+const validateCity = (city) => city.match(/^[A-Za-z]+$/);
+
+const validatePhone = (phone) => phone.match(/^[0-9]+$/);
+const validateZip = (zip) => zip.match(/^[0-9]+$/);
```
Comment (bejanaf) [inline client/src/lib/Validation.js:10]: keep in mind uk zip codes includes letters as well ;)

File: client/src/pages/AddForm.js [hunk 2]
```diff
@@ -71,10 +70,14 @@ export default function AddForm({ onAddOrganizationsAndAnimals }) {
       onAddOrganizationsAndAnimals(organization, animal);
       setOrganization(initialOrganization);
       setAnimal(initialAnimal);
+      setIsSuccessView(true);
       setIsError(false);
+      setIsSuccessView(true);
```
Comment (bejanaf) [inline client/src/pages/AddForm.js:74]: you call this function twice, please delete one :)

File: client/src/pages/AddForm.js [hunk 3]
```diff
@@ -119,236 +122,87 @@ export default function AddForm({ onAddOrganizationsAndAnimals }) {
...
+              animal={animal}
+              onUpdateAnimal={updateAnimal}
+              onRemoveTrait={removeTrait}
+              hiddenFileInput={hiddenFileInput}
+              imageName={imageName}
+              onUpdateImage={updateImage}
+              onHandleClick={handleClick}
+              onUpdateTraits={updateTraits}
+            />
+            {isError ? (
```
Comment (bejanaf) [inline client/src/pages/AddForm.js]: consider using nullish coaleascing operator then you don't need null

**Review by bejanaf — CHANGES_REQUESTED** [review]
please review the comments i made :)

**Comment by SabMueller** [comment]
everything done :)

**Review by bejanaf — APPROVED** [review]

**Pull request merged**